automatic_push/rfc_push.py

```diff
@@ -0,0 +1,136 @@
+
+
+@dataclass
+class PullRequestCreationResult:
+    pr_created: bool
+    message: str
+
+
+def create_new_rfcs_pull_request(
+    files_to_update: list[str],
+    forked_repo: ModifiableRepoUtil,
+    logger: logging.Logger,
+    config: ConfigParser = create_config(),
+) -> PullRequestCreationResult:
+    if not files_to_update:
+        return PullRequestCreationResult(False, 'No files to update')
+    token = config.get('Secrets-Section', 'yang-catalog-token')
+    username = config.get('General-Section', 'repository-username')
+    repo_previous_active_branch = forked_repo.repo.active_branch
+    try:
+        forked_repo.repo.git.checkout('HEAD', b=datetime.now().strftime('%Y_%m_%d_%H_%M_%S'))
+        _extract_modules(files_to_update, forked_repo, config)
+        if not forked_repo.repo.untracked_files:
+            forked_repo.repo.git.checkout(repo_previous_active_branch)
+            return PullRequestCreationResult(False, 'No files to update after extraction')
+        result = _create_pull_request(forked_repo, username, token, logger, config)
+        forked_repo.repo.git.checkout(repo_previous_active_branch)
+        return result
+    except Exception as e:
+        logger.exception('Unexpected error occurred during an automatic PullRequest creation')
+        forked_repo.repo.git.checkout(repo_previous_active_branch)
+        return PullRequestCreationResult(False, f'Unexpected error\n{e}')
+
+
+def _extract_modules(files_to_update: list[str], repo: ModifiableRepoUtil, config: ConfigParser):
+    private_dir = config.get('Web-Section', 'private-directory')
+    extract_to = os.path.join(repo.local_dir, 'standard/ietf/RFCtemp')
+    rfc_directory = os.path.join(repo.local_dir, 'standard/ietf/RFC')
+    if os.path.exists(extract_to):
+        shutil.rmtree(extract_to)
+    os.makedirs(extract_to)
+    with open(os.path.join(private_dir, 'IETFYANGRFC.json'), 'r') as f:
+        rfcs_dict = json.load(f)
+    extracted_rfcs = set()
+    cwd = os.getcwd()
+    os.chdir(rfc_directory)
+    for filename in files_to_update:
+        rfc_url = rfcs_dict.get(filename)
+        if not rfc_url:
+            continue
+        try:
+            rfc_url = rfc_url.split('<a href=\"')[1].split('\">')[0]
+        except IndexError:
+            continue
+        if rfc_url not in extracted_rfcs:
+            xym.xym(source_id=rfc_url, srcdir='', dstdir=extract_to)
+            extracted_rfcs.add(rfc_url)
+        filename_without_revision = f'{filename.split("@")[0]}.yang'
+        extracted_file_path = os.path.join(extract_to, filename_without_revision)
+        if not os.path.exists(extracted_file_path):
+            continue
+        shutil.copy2(extracted_file_path, filename)
+        if os.path.islink(filename_without_revision):
+            os.unlink(filename_without_revision)
+        os.symlink(filename, filename_without_revision)
+    os.chdir(cwd)
+    shutil.rmtree(extract_to)
+
+
+def _create_pull_request(
```

Comment:
I would also think about moving part of this method into repoutil. At least the code responsible for creating PR.
Try to compare it with code in `draftPull.py` local and create method which will get necessary info as parameter